lib/utils.js

```diff
@@ -3,7 +3,30 @@
  */
 
 export function clone ( source ) {
-  return source ? JSON.parse( JSON.stringify( source ) ) : {};
```

Comment:
We need to make sure source is set, to avoid crashing when used with server-side rendering. I know it is caused by "bad code" by the user, but I see no reason to NOT make this check.